src/sth_database.js

```diff
@@ -236,6 +236,227 @@
+   * @param {string} attributeId The updated attribute id
+   * @param {string} attributeType The updated attribute type
+   * @param {string} attributeValue The updated attribute value
+   * @param {Function} callback Function to call once the operation completes
+   */
+  function storeAggregatedData4Resolution(
+    collectionName4Aggregated, resolution, timestamp, entityId, entityType,
+    attributeId, attributeType, attributeValue, callback) {
+    /*
+     Currently the MongoDB $ positional update operator cannot be combined with upserts
+      (see http://docs.mongodb.org/manual/reference/operator/update/positional/#upsert).
+      This issue is known and currently under study: https://jira.mongodb.org/browse/SERVER-3326
+      Once the issue is solved, it will be possible to prepopulate collections or update their docs
+      using just one update operation like this:
+      getAggregatedModel(collectionName4Aggregated).update(
+        // Returning all the update operators currently returned by getAggregateUpdate4Insert
+        //  and getAggregateUpdate4Update in the same object
+        getAggregateUpdateCondition(resolution, timestamp),
+        getAggregateUpdate(attributeValue, resolution, timestamp),
+        {upsert: true},
+        function (err) {
+          callback(err);
+        }
+      );
+     */
+    // Prepopulate the aggregated data collection if there is no entry for the concrete
+    //  origin, resolution and range.
+    getAggregatedModel(collectionName4Aggregated).update(
+      getAggregateUpdateCondition(resolution, timestamp),
+      getAggregateUpdate4Insert(attributeValue, resolution, timestamp),
+      {upsert: true},
+      function (err) {
+        // Once the aggregated data collection has been prepopulated (if needed),
+        //  aggregate the new value received.
+        getAggregatedModel(collectionName4Aggregated).update(
+          getAggregateUpdateCondition(resolution, timestamp),
+          getAggregateUpdate4Update(attributeValue, resolution, timestamp),
+          function (err) {
+            if (callback) {
+              callback(err);
+            }
+          }
+        );
+      }
+    );
+  }
+
+  /**
+   * Stores the aggregated data for a new event (attribute value)
+   * @param {Date} timestamp The date the event arrived
+   * @param {string} entityId The entity id associated to updated attribute
+   * @param {string} entityType The entity type associated to the updated attribute
+   * @param {string} attributeId The updated attribute id
+   * @param {string} attributeType The updated attribute type
+   * @param {string} attributeValue The updated attribute value
+   * @param {Function} callback Function to call once the operation completes
+   */
+  function storeAggregatedData(
+    timestamp, entityId, entityType, attributeId, attributeType, attributeValue, callback) {
+    var collectionName4Aggregated = getCollectionName4Aggregated(entityId, attributeId);
```

Comment:
This seems to be creating a collection name based on the entityId and the attributeId... this should be aligned (later or now, I don't know) with the naming conventions we are discussing offline.